clang3.9.1 util.c - comparison of unsigned expression < 0 is always false

**Environment**

Fedora 26 x86_64
clang 3.9.1
git faef420b1e4eecee4e2718083ee5bc11d22e9c1e

**Error**

```
/usr/bin/clang -DHAVE_CONFIG_H -I. -I..    -Wall -Werror -DVERSIONCOMMIT=\""0.8.1, commit: `git describe --dirty --always --tags 2> /dev/null || true`"\" -g -O2 -MT util.o -MD -MP -MF $depbase.Tpo -c -o util.o util.c &&\
mv -f $depbase.Tpo $depbase.Po
util.c:251:9: error: comparison of unsigned expression < 0 is always false [-Werror,-Wtautological-compare]
                if (l < 0) {
                    ~ ^ ~
1 error generated.
make[2]: *** [Makefile:362: util.o] Error 1
```

--- a/src/util.c
+++ b/src/util.c
@@ -239,7 +239,7 @@ int hyper_getgrouplist(const char *user, gid_t group, gid_t *groups, int *ngroup
 
 int hyper_write_file(const char *path, const char *value, size_t len)
 {
-	size_t size = 0, l;
+	ssize_t size = 0, l;
 	int fd = open(path, O_WRONLY);
 	if (fd < 0) {
 		perror("open file failed");

